### ISSUE TITLE
Did pyright drop support for `pyrightconf.json` and only uses `.pyrightconfig.json`?

### ISSUE THREAD

**Issue by cmac4603** [microsoft/pyright#4464]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
All imports from 3rd party modules could not be found. This bug was found specifically using `venvPath` & `venv`. Running the following command fixed pyright for me.
```bash
mv pyrightconfig.json .pyrightconfig.json
```

**To Reproduce**
Steps to reproduce the behavior.

**Expected behavior**
This definitely worked for me before as `pyrightconfig.json`. Moving to `.pyrightconfig.json` confirmed to me that this was the issue.

**VS Code extension or command-line**
Neovim

**Additional context**

**Comment by cmac4603** [comment]
```
› neofetch             
                    'c.          cmacrae@HMILSQ-W34ELVCF 
                 ,xNMM.          ----------------------- 
               .OMMMMo           OS: macOS 12.6.2 21G320 x86_64 
               OMMM0,            Host: MacBookPro15,1 
     .;loddo:' loolloddol;.      Kernel: 21.6.0 
   cKMMMMMMMMMMNWMMMMMMMMMM0:    Uptime: 4 days, 5 hours, 3 mins 
 .KMMMMMMMMMMMMMMMMMMMMMMMWd.    Packages: 41 (brew) 
 XMMMMMMMMMMMMMMMMMMMMMMMX.      Shell: zsh 5.8.1 
;MMMMMMMMMMMMMMMMMMMMMMMM:       Resolution: 1680x1050 
:MMMMMMMMMMMMMMMMMMMMMMMM:       DE: Aqua 
.MMMMMMMMMMMMMMMMMMMMMMMMX.      WM: Quartz Compositor 
 kMMMMMMMMMMMMMMMMMMMMMMMMWd.    WM Theme: Blue (Dark) 
 .XMMMMMMMMMMMMMMMMMMMMMMMMMMk   Terminal: tmux 
  .XMMMMMMMMMMMMMMMMMMMMMMMMK.   CPU: Intel i7-9750H (12) @ 2.60GHz 
    kMMMMMMMMMMMMMMMMMMMMMMd     GPU: Intel UHD Graphics 630, Radeon Pro 555X 
     ;KMMMMMMMWXXWMMMMMMMk.      Memory: 11248MiB / 16384MiB 
       .cooc,.    .,coo:.
                                                         
                                                         
```

**Comment by erictraut** [comment]
Pyright's configuration file must be named `pyrightconfig.json`. It won't be recognized if you place a `.` before the name. This hasn't changed.

**Comment by cmac4603** [comment]
Apologies, title was incorrect, but the command in the description was correct to get this to work:
```bash
mv pyrightconfig.json .pyrightconfig.json
```
In fact, it now does work since I ammended `pyrightconfig.json` to `.pyrightconfig.json`. If anyone could provide some insight that would be awesome

**Comment by cmac4603** [comment]
Unless it's reading from the terminal env where I had already activated the virtualenv... 🤔

**Comment by erictraut** [comment]
By renaming the config file, pyright is no longer treating it as a config file, so pyright uses its default behaviors.

By default, pyright uses the currently-configured python environment. It does so by executing `python3` (or `python` on Windows) and running a small script that dumps the `sys.path` import resolution paths. That probably explains the behavior you're seeing.

**Comment by cmac4603** [comment]
I've tested this with a virtualenv activated and deactivated before starting neovim and pyright only works with `.pyrightconfig.json`. I'm not crazy.
```
nvim --version
NVIM v0.8.2
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by brew@Monterey
```
